kudos-services/src/main/java/org/exoplatform/kudos/listener/KudosActivityListener.java

```diff
@@ -46,11 +54,31 @@ public void updateActivity(ActivityLifeCycleEvent activityLifeCycleEvent) {
     }
   }
 
+  @Override
+  public void deleteActivity(ActivityLifeCycleEvent activityLifeCycleEvent) {
+    ExoSocialActivity activity = activityLifeCycleEvent.getSource();
+    List<Kudos> linkedKudosList = kudosService.getKudosListOfActivity(activity.getId());
+    if (!StringUtils.equals(activity.getType(), KUDOS_ACTIVITY_COMMENT_TYPE) && linkedKudosList.isEmpty()) {
+      return;
+    }
+    deleteLinkedKudos(linkedKudosList);
```

Comment:
Does this the same thing ?

```suggestion
    if (linkedKudosList.isEmpty()) {
      deleteLinkedKudos(linkedKudosList);
    }
```

Reply:
yes, right!